File: Doubly_linked_list/doubly_linked_list_insert_end.c
// File Name: doubly_linked_list_insert_end.c
// Program to insert a node at the end of a doubly linked list

#include <stdio.h>
#include <stdlib.h>

// Definition of node
struct node {
    int data;
    struct node *prev;
    struct node *next;
};

struct node *head = NULL;

// Function to insert at end
void insert_end(int value) {
    struct node *newnode, *temp;

    newnode = (struct node *)malloc(sizeof(struct node));
    newnode->data = value;
    newnode->prev = NULL;
    newnode->next = NULL;

    // If list is empty
    if (head == NULL) {
        head = newnode;
        return;
    }

    temp = head;
    while (temp->next != NULL)
        temp = temp->next;

    temp->next = newnode;
    newnode->prev = temp;
}

int main() {
    insert_end(10);
    return 0;
}

File: Doubly_linked_list/DoublyLinkedListInsertEnd.java
// File Name: DoublyLinkedListInsertEnd.java
// Program to insert a node at the end of a doubly linked list

class Node {
    int data;
    Node prev;
    Node next;

    Node(int data) {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

public class DoublyLinkedListInsertEnd {
    static Node head = null;

    // Function to insert at end
    static void insert_end(int value) {
        Node newNode = new Node(value);

        // If list is empty
        if (head == null) {
            head = newNode;
            return;
        }

        Node temp = head;
        while (temp.next != null)
            temp = temp.next;

        temp.next = newNode;
        newNode.prev = temp;
    }

    public static void main(String[] args) {
        insert_end(10);
    }
}
